frontend/javascripts/oxalis/model/helpers/compaction/compact_toggle_actions.ts

```diff
@@ -108,77 +129,113 @@ function findCommonAncestor(
   return _.last(commonPath);
 }
 
-function isCommonAncestorToggler(
-  skeletonTracing: SkeletonTracing,
+function isCommonAncestorToggler<T extends SkeletonTracing | VolumeTracing>(
+  tracing: T,
   commonAncestor: number | undefined,
-): [boolean, Tree[], number] {
-  const groupToTreesMap = createGroupToTreesMap(skeletonTracing.trees);
-  const groupWithSubgroups = getGroupByIdWithSubgroups(skeletonTracing.treeGroups, commonAncestor);
-  const allTreesOfAncestor: Tree[] =
-    groupWithSubgroups.length === 0
-      ? _.values(skeletonTracing.trees)
-      : _.flatMap(groupWithSubgroups, (groupId: number): Tree[] => groupToTreesMap[groupId] || []);
+): [boolean, Array<T extends SkeletonTracing ? Tree : Segment>, number] {
+  let allItemsOfAncestor: Array<Tree | Segment> = [];
+  if (tracing.type === "skeleton") {
+    const items = tracing.trees;
+    const groups = tracing.treeGroups;
+    const groupToTreesMap = createGroupToTreesMap(items);
+    const groupWithSubgroups = getGroupByIdWithSubgroups(groups, commonAncestor);
+    allItemsOfAncestor =
+      groupWithSubgroups.length === 0
+        ? _.values(items)
+        : _.flatMap(
+            groupWithSubgroups,
+            (groupId: number): Tree[] => groupToTreesMap[groupId] || [],
+          );
+  } else {
+    const items = tracing.segments;
+    const groups = tracing.segmentGroups;
+    const groupToTreesMap = createGroupToSegmentsMap(items);
+    const groupWithSubgroups = getGroupByIdWithSubgroups(groups, commonAncestor);
+    allItemsOfAncestor =
+      groupWithSubgroups.length === 0
+        ? Array.from(items.values())
```

Comment:
Note that segments use a DiffableMap whereas the trees only use a simple Object. this could be unified. I created https://github.com/scalableminds/webknossos/issues/8596